### ISSUE TITLE
Nested and long tuples  lead to crash

### ISSUE THREAD

**Issue by deemytch** [ufirstgroup/ymlr#117]
Tuples with more than two elements seems leads to crash.

```elixir
iex> {127, 0, 0, 1} |> Ymlr.document!
(ArgumentError) The given data {127, 0, 0, 1} cannot be converted to YAML.

iex> {127, 0, 0, 1} |> JSON.encode!  
"[127,0,0,1]"

iex> {127,0} |>  Ymlr.document!()
"---\n# 127\n0\n"
iex> Ymlr.document!({1,2,3})
** (ArgumentError) The given data {1, 2, 3} cannot be converted to YAML.
    (ymlr 3.0.1) lib/ymlr/encoder.ex:114: Ymlr.Encoder.encode_as_io_list/2
    (ymlr 3.0.1) lib/ymlr/encoder.ex:47: Ymlr.Encoder.to_s!/1
    (ymlr 3.0.1) lib/ymlr.ex:34: Ymlr.document!/1
    iex:31: (file)
```

I think without tuples supporting Ymlr is completely unusable for real data.

One more example.

```elixir
%{
  bindings: %{},
  body_length: 0,
  cert: :undefined,
  charset: :undefined,
  has_body: false,
  headers: %{
    "accept" => "*/*",
    "host" => "localhost:3700",
    "user-agent" => "curl/8.0.1"
  },
  host: "localhost",
  host_info: :undefined,
  media_type: {"text", "plain"},
  method: "GET",
  path: "/rest",
  path_info: :undefined,
  peer: {{127, 0}, 57276},
  pid: nil,
  port: 3700,
  qs: "",
  ref: :cowboy_elixir,
  resp_headers: %{
    "content-type" => ["text", "/", "plain", ""],
    "vary" => ["accept"]
  },
  scheme: "http",
  sock: {{127, 0}, 3700},
  streamid: 1,
  version: :"HTTP/1.1"
}
iex(34)> |> Ymlr.document!()
** (ArgumentError) The given data {"text", "plain"} cannot be converted to YAML.
    (ymlr 3.0.1) lib/ymlr/encoder.ex:114: Ymlr.Encoder.encode_as_io_list/2
    (ymlr 3.0.1) lib/ymlr/encoder.ex:93: anonymous fn/3 in Ymlr.Encoder.encode_as_io_list/2
    (elixir 1.14.0) lib/enum.ex:1658: Enum."-map/2-lists^map/1-0-"/2
    (elixir 1.14.0) lib/enum.ex:1658: Enum."-map/2-lists^map/1-0-"/2
    (ymlr 3.0.1) lib/ymlr/encoder.ex:87: Ymlr.Encoder.encode_as_io_list/2
    (ymlr 3.0.1) lib/ymlr/encoder.ex:47: Ymlr.Encoder.to_s!/1
    (ymlr 3.0.1) lib/ymlr.ex:34: Ymlr.document!/1
    iex:34: (file)
```

**Comment by mruoss** [comment]
That's true. Tuple's aren't supported at this moment.

**Comment by mruoss** [comment]
Will be addressed in #118
